core/chains/evm/evm_txm.go

```diff
@@ -0,0 +1,49 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/smartcontractkit/sqlx"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmconfig "github.com/smartcontractkit/chainlink/core/chains/evm/config"
+	httypes "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+var _ httypes.HeadTrackable = &evmTxm{}
+
+// evmTxm is an evm wrapper over the generic TxManager interface
+type evmTxm struct {
+	httypes.HeadTrackable
+	txmgr.TxManager
+}
+
+func (e evmTxm) OnNewLongestChain(ctx context.Context, head *evmtypes.Head) {
+	e.TxManager.OnNewLongestChain(ctx, head)
+}
```

Comment:
I am not sure why this is necessary but I believe you that it is.

Reply:
Same here. I don't know why it doesn't work without this, but if I remove this function, I see these errors:

FAIL	github.com/smartcontractkit/chainlink/core/chains/evm [build failed]
# github.com/smartcontractkit/chainlink/core/chains/evm [github.com/smartcontractkit/chainlink/core/chains/evm/client.test]
./evm_txm.go:16:31: cannot use &evmTxm{} (value of type *evmTxm) as "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types".HeadTrackable value in variable declaration: *evmTxm does not implement "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types".HeadTrackable (missing method OnNewLongestChain)
./chain.go:120:28: cannot use txm (variable of type *evmTxm) as "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types".HeadTrackable value in argument to headBroadcaster.Subscribe: *evmTxm does not implement "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types".HeadTrackable (missing method OnNewLongestChain)

Reply:
Huh, weird. Could it have been a pointer issue? I'm surprised to see the receiver here is not a pointer.

Reply:
I didn't understand your question. 

evmTxm implements both interfaces:
- 	httypes.HeadTrackable. (requires OnLongestChain(..., *evmtypes.Head))
- txmgr.TxManager (requires OnLongestChain(..., txmgrtypes.Head))

I assumed just letting this OnLongestMethod inheriting from TxManager would work, but it doesn't. I have to create above function to pass-through.

Reply:
From my understanding of go, this shouldn't be necessary. Can we figure out why it's here?

Reply:
This method is not necessary. This is because the type assertion is generic instead of specifying `[*evmtypes.Head]`.

Reply:
Er, the concrete type with the same name is not connected to the generic one. This should be removable if they are derived from one another.

Reply:
Actually, this whole type is not necessary: https://github.com/smartcontractkit/chainlink/pull/8744